### DISCUSSION TITLE
hover_swarm.launch and mocap_helper.launch does not output Motive data

### DISCUSSION THREAD

**Discussion by pthinesh** [USC-ACTLab/crazyswarm#384] [Q&A] [answered]
I am trying to fly one Crazyflie using hover_swarm.launch with OptiTrack motion capture system. Everything works fine, except that rviz not showing the Crazyflie movements.
There were no errors in the terminal. I also tried mocap_helper.launch, there are no position updates printed.

I made the following modifications in "hover_swarm.launch" file:
      **motion_capture_type: "optitrack" # one of none,vicon,optitrack,qualisys,vrpn
      object_tracking_type: "motionCapture" # one of motionCapture,libobjecttracker**
      send_position_only: False # set to False to send position+orientation; set to True to send position only
      # vicon_host_name: "vicon"
      **optitrack_host_name: "192.168.1.1" # Windows (Motive) IP address**
      # qualisys_host_name: "10.0.5.219"
      # qualisys_base_port: 22222
      # vrpn_host_name: "192.168.1.1"
      **optitrack_local_ip: "192.168.1.3" # Ubuntu IP address
      optitrack_server_ip: "192.168.1.1" # Windows (Motive) IP address**

Motive Streaming settings:
![Capture2](https://user-images.githubusercontent.com/84022671/117948070-c1300180-b34b-11eb-97e2-7cc34f6b841d.PNG)

Terminal output when running hello_world.py with hover_swarm.launch:
(Crazyflie takes off and lands without any error)
![Screenshot from 2021-05-12 18-07-34](https://user-images.githubusercontent.com/84022671/117949404-0e60a300-b34d-11eb-9a4b-4c132a20dcc1.png)

Terminal output when running mocap_helper.launch:
![Screenshot from 2021-05-12 18-10-16](https://user-images.githubusercontent.com/84022671/117949694-5b447980-b34d-11eb-9ffa-9c4080ae52e0.png)

I'm using ROS noetic on Ubuntu 20.04.
Motive version is 2.1.2
Windows and Ubuntu are connected using a separate Ethernet port, and there are no problem with the network (motion data are successfully received when using VRPN).

**Comment by whoenig** [comment]
Could you try running https://github.com/whoenig/NatNetSDKCrossplatform and report if you can receive data that way?

**Comment by pthinesh** [comment]
Hi @whoenig, Thanks for the reply.

I tried 'NatNetSDKCrossplatform'; following is the output
![Screenshot from 2021-05-13 11-52-40](https://user-images.githubusercontent.com/84022671/118071018-ea9b6c80-b3e1-11eb-869f-05a008079a07.png)

Then again I tried VRPN (it works without any error)
![Screenshot from 2021-05-13 11-53-03](https://user-images.githubusercontent.com/84022671/118071072-030b8700-b3e2-11eb-937a-b8f15ad53416.png)

**Comment by pthinesh** [comment]
Hi @whoenig, I tried disconnecting the Wifi and run hover_swarm.launch, following error occurred;
![Screenshot from 2021-05-13 12-32-33](https://user-images.githubusercontent.com/84022671/118073716-5d5b1680-b3e7-11eb-9b51-5cf7eefef18b.png)

VRPN works fine.
I couldn't find the mistake, could you help me resolve this?

**Comment by whoenig** [comment, marked as answer]
See https://github.com/whoenig/NatNetSDKCrossplatform/issues/2.

**Comment by Sherry-97** [comment]
@pthinesh I seem to have the same problem as you, does your problem is solved？

**Reply by pthinesh** [reply]
@Sherry-97 Yes, I followed the instructions from https://github.com/whoenig/NatNetSDKCrossplatform/issues/2#issuecomment-841666150
